rpm/0033-sailfishos-configure-Read-rustc-host-from-environmen.patch

```diff
@@ -0,0 +1,109 @@
+\ No newline at end of file
++{"files":{"Cargo.lock":"3aff5f8b0a7f4d72852b11b0526f0002e6bf55f19f1ebd6470d7f97fbd540e60","Cargo.toml":"6ab10d9b6a9c6f0909074e6698c90c6b6a7223661ec2e83174d2593117cbe7f2","LICENSE-APACHE":"a60eea817514531668d7e00765731449fe14d059d3249e0bc93b36de45f759f2","LICENSE-MIT":"378f5840b258e2779c39418f3f2d7b2ba96f1c7917dd6be0713f88305dbda397","README.md":"7184fbdf375a057e673257348f6d7584c0dd11b66318d98f3647f69eb610b097","src/bin/gcc-shim.rs":"b77907875029494b6288841c3aed2e4939ed40708c7f597fca5c9e2570490ca6","src/com.rs":"bcdaf1c28b71e6ef889c6b08d1ce9d7c0761344a677f523bc4c3cd297957f804","src/lib.rs":"dff2f917bc733c39e4b87cb195db36ed02f2e37487adeddf66d7164975e6e94e","src/registry.rs":"3cc1b5a50879fa751572878ae1d0afbfc960c11665258492754b2c8bccb0ff5d","src/setup_config.rs":"7014103587d3382eac599cb76f016e2609b8140970861b2237982d1db24af265","src/winapi.rs":"ea8b7edbb9ff87957254f465c2334e714c5d6b3b19a8d757c48ea7ca0881c50c","src/windows_registry.rs":"388e79dcf3e84078ae0b086c6cdee9cf9eb7e3ffafdcbf3e2df26163661f5856","tests/cc_env.rs":"e02b3b0824ad039b47e4462c5ef6dbe6c824c28e7953af94a0f28f7b5158042e","tests/cflags.rs":"57f06eb5ce1557e5b4a032d0c4673e18fbe6f8d26c1deb153126e368b96b41b3","tests/cxxflags.rs":"c2c6c6d8a0d7146616fa1caed26876ee7bc9fcfffd525eb4743593cade5f3371","tests/support/mod.rs":"71620b178583b6e6e5e0d4cac14e2cef6afc62fb6841e0c72ed1784543abf8ac","tests/test.rs":"1605640c9b94a77f48fc92e1dc0485bdf1960da5626e2e00279e4703691656bc"},"package":"aa87058dce70a3ff5621797f1506cb837edd02ac4c0ae642b4542dce802908b8"}
+diff --git a/third_party/rust/cc/src/lib.rs b/third_party/rust/cc/src/lib.rs
+index 621d31d6b8a9..61e9d631fbb9 100644
+--- a/third_party/rust/cc/src/lib.rs
++++ b/third_party/rust/cc/src/lib.rs
+@@ -1634,6 +1634,7 @@ impl Build {
+ 
+     fn has_flags(&self) -> bool {
+         let flags_env_var_name = if self.cpp { "CXXFLAGS" } else { "CFLAGS" };
++        println!("has_flags name {:?}!!!", &flags_env_var_name);
+         let flags_env_var_value = self.get_var(flags_env_var_name);
+         if let Ok(_) = flags_env_var_value {
+             true
+@@ -1896,7 +1897,8 @@ impl Build {
+         let default = if host.contains("solaris") {
+             gnu
+         } else {
+-            traditional
++            //traditional
++            "armv7-unknown-linux-gnueabihf"
```

Comment:
Are we sure this works when building for aarch64?

Reply:
No, not sure at all. But I'm also not 100% sure whether this is necessary or not. It'll need investigating.

Reply:
I've overlaid a patch to reverse this change, so we can see whether it still builds without it.

Reply:
Just in case it's of interest: I also tried dropping this entire patch, but the build then failed on arm, so I brought it back.